utils/fixed_decimal/src/integer.rs

```diff
@@ -0,0 +1,99 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use core::convert::TryFrom;
+use core::fmt;
+
+use core::str::FromStr;
+
+use crate::Error;
+use crate::FixedDecimal;
+
+/// A [`FixedInteger`] is a [`FixedDecimal`] with no fractional part.
+///
+///
+/// # Examples
+///
+/// ```
+/// # use std::convert::TryFrom;
+/// # use std::str::FromStr;
+/// use fixed_decimal::Error;
+/// use fixed_decimal::FixedDecimal;
+/// use fixed_decimal::FixedInteger;
+///
+/// assert_eq!(
+///     FixedInteger::try_from(FixedDecimal::from(5)),
+///     Ok(FixedInteger::from(5))
+/// );
+/// assert_eq!(
+///     FixedInteger::try_from(FixedDecimal::from_str("05").unwrap()),
+///     Ok(FixedInteger::from_str("05").unwrap())
+/// );
+/// assert_eq!(
+///     FixedInteger::try_from(FixedDecimal::from_str("5.0").unwrap()),
+///     Err(Error::Limit)
+/// );
+/// ```
+#[derive(Debug, Clone, PartialEq, Default)]
+pub struct FixedInteger(FixedDecimal);
+
+impl<'a> From<&'a FixedInteger> for &'a FixedDecimal {
+    fn from(value: &'a FixedInteger) -> Self {
+        &value.0
+    }
+}
```

Comment:
Nit: This impl puts in the public API that a `&FixedInteger` must be able to return a `&FixedDecimal`. This is a strong statement that I prefer not to make, because it prevents us from changing the data model of `FixedInteger` in the future. I prefer to make it `impl From<FixedInteger> for FixedDecimal`.

However, _within this crate_, you can (and should) use the actual FixedDecimal by just accessing the `.0` field.